### PR TITLE
remove outdated write-up against sudo passwords

### PR DESCRIPTION
**Pull request by adrelanos**
Maybe not perfect but better than spreading highly outdated, since revised viewpoints.

Fixes https://github.com/QubesOS/qubes-issues/issues/8823

### CODE REVIEW

**Comment by marmarek** [comment]
I don't think it's appropriate to remove this description, maybe just rephrase to not suggest it's directly from some file in a VM that isn't there anymore. After all, password-less root access in VM is still the default and it should be documented why it's the case.

**Comment by adrelanos** [comment]
But the new state is more correct than the current state which is a huge reasoning pro passwordless sudo which has now been revised?

I believe that this might have [created](https://forums.whonix.org/t/benefiting-from-security-misc-inside-qubes-whonix/18124) the impression that Qubes isn't interested in sudo hardening, which I think is no longer true.

So rather than being completely anti-sudo password I think it's better to just link to the ticket so at least highly technical users can find out.

> is still the default and it should be documented why it's the case.

It's the case because nobody implemented https://github.com/QubesOS/qubes-issues/issues/2695 yet?

**Comment by marmarek** [comment]
> So rather than being completely anti-sudo password I think it's better to just link to the ticket so at least highly technical users can find out.

Well, all those points about isolation (or rather lack of it) a user _with shell/arbitrary exec access_ from root (if the user can actually get root access, via some extra authorization) are still very much true. If you can use `sudo` with some extra steps, then malicious code on your user account can simply wait until you perform those extra steps, whatever they are.

So, most of this description is still accurate. The thing that changed is we made the password-less root access opt-out, if you have a case where you more or less never use sudo at all. In fact, uninstalling passwordless-root package prevents using sudo for arbitrary commands at all (unless you configure some alternative authorization).

> > is still the default and it should be documented why it's the case.
> 
> It's the case because nobody implemented [QubesOS/qubes-issues#2695](https://github.com/QubesOS/qubes-issues/issues/2695) yet?

There is `qvm-console-dispvm` that can be used to access root account even if sudo is disabled. So, some alternative method exists already, if one really wants to go this route.

But again, the current state is the passwordless-root enabled by default, and it's wrong to remove documentation explaining why it is the case. It might be appropriate if that default changes at some point.

**Comment by adrelanos** [comment]
> If you can use `sudo` with some extra steps, then malicious code on your user account can simply wait until you perform those extra steps, whatever they are.

I would suggest to have VMs that by user volunteer policy (or one day even a Qubes dom0 VM setting) can never get sudo. Useful for VMs only used for browsers.

> So, most of this description is still accurate. The thing that changed is we made the password-less root access opt-out, if you have a case where you more or less never use sudo at all. In fact, uninstalling passwordless-root package prevents using sudo for arbitrary commands at all (unless you configure some alternative authorization).

Good.

> > > is still the default and it should be documented why it's the case.
> > 
> > 
> > It's the case because nobody implemented [QubesOS/qubes-issues#2695](https://github.com/QubesOS/qubes-issues/issues/2695) yet?
> 
> There is `qvm-console-dispvm` that can be used to access root account even if sudo is disabled. So, some alternative method exists already, if one really wants to go this route.

Nice. Also needs documentation.

> But again, the current state is the passwordless-root enabled by default, and it's wrong to remove documentation explaining why it is the case. It might be appropriate if that default changes at some point.

Ok.

**Comment by SaswatPadhi** [comment]
> But again, the current state is the passwordless-root enabled by default

It doesn't seem to be the default for any of the  minimal VMs provided by Qubes.

**Comment by runephilosof-abtion** [comment]
@andrewdavidwong You added the "waiting for author" label, so I guess you want @adrelanos to respond before reviewing again?

If the outdated reasoning for adding passwordless in the first place must stay, I would prefer to prepend it with:
> This is the original reasoning for implementing passwordless sudo, see issue https://github.com/QubesOS/qubes-issues/issues/2695 for revising the policy.

But I would prefer to just have it deleted as in this PR.

> the current state is the passwordless-root enabled by default, and it's wrong to remove documentation explaining why it is the case

The proposal links to a place where that explanation can be found. So because of that I see nothing wrong with this PR. It is a better state than the current.

**Comment by unman** [comment]
There's confusion here.
For most Qubes users, in default set-up, passwordless root is enabled.
As far as I am aware, the thinking behind this has NOT changed, and no one has given any reason to think that it has.
Given that the explanation in the qubes has been (unfortunately) removed in 4.2 the PR should be revised accordingly.

The fact that passwordless root is not enabled in minimal templates is well documented.
The possibility of using dom0 authorisation is linked from this page.

**Comment by adrelanos** [comment]
meta: https://github.com/QubesOS/qubes-issues/issues/3629#issuecomment-1935589503

**Comment by unman** [comment]
This PR is based on a mistaken view - that the original page contains "highly outdated, since revised viewpoints"

Passwordless root remains the default, and as @marmarek has said, most of the description is still accurate.
I am moving details of the implementation to developer docs (as per discussion in PR #1296), and updating this page.

**Comment by tlaurion** [comment]
@andrewdavidwong ELI5?

**Comment by unman** [comment]
*None* of the documentation is ELI5
I thought the "rant" was already pitched quite low.
Can you suggest what you like to see?
